### PR TITLE
This introduces a WIP "export" using the logic from CNAB GO

### PR DESCRIPTION
**Pull request by jeremyrickard**
# What does this change

This introduces some initial plumbing for generating a "thick" bundle. 

# What issue does it fix
Part of #544

_If there is not an existing issue, please make sure we have context on why this change is needed. See our Contributing Guide for [examples of when an existing issue isn't necessary][1]._

[1]: https://github.com/deislabs/porter/blob/master/CONTRIBUTING.md#when-to-open-a-pull-request

# Notes for the reviewer

Note: it doesn't use all of cnab-go yet, i copied some code over pending a refactor in
cnab-go. I'd like to merge this more or less as is so we can start experimenting with the flip side of the equation.

### CODE REVIEW

**Comment by carolynvs** [comment]
[fix.patch.txt](https://github.com/deislabs/porter/files/3691150/fix.patch.txt)

Here is the fix to the Makefile and lock that should get the build passing. Sorry about that, this was a break from another PR that I missed...

File: pkg/porter/archive.go [hunk 1]
```diff
@@ -2,24 +2,172 @@ package porter
...
 
 // Archive is a composite function that generates a CNAB thick bundle. It will pull the invocation image, and
 // any referenced images locally (if needed), export them to individual layers, generate a bundle.json and
 // then generate a gzipped tar archive containing the bundle.json and the images
 func (p *Porter) Archive(opts ArchiveOptions) error {
-	return fmt.Errorf("archive is not yet implemented")
+
+	err := p.prepullBundleByTag(&opts.BundleLifecycleOpts)
+	if err != nil {
+		return errors.Wrap(err, "unable to pull bundle before invoking credentials generate")
```
Comment (vdice) [inline pkg/porter/archive.go]: Looks like the error message needs updating here

File: pkg/porter/archive.go [hunk 2]
```diff
@@ -2,24 +2,172 @@ package porter
...
+		out:                   p.Config.Out,
+		logs:                  p.Config.Out,
+		bundle:                bun,
+		destination:           dest,
+		imageStoreConstructor: ctor,
+	}
+	if err := exp.export(); err != nil {
+		return err
+	}
+	// if ex.verbose {
```
Comment (vdice) [inline pkg/porter/archive.go]: Should we just remove this section?

File: cmd/porter/bundle.go [hunk 3]
```diff
@@ -303,5 +303,7 @@ func buildBundleArchiveCommand(p *porter.Porter) *cobra.Command {
 	f := cmd.Flags()
 	f.StringVarP(&opts.File, "file", "f", "", "Path to the Porter manifest. Defaults to `porter.yaml` in the current directory.")
 	f.StringVar(&opts.CNABFile, "cnab-file", "", "Path to the CNAB bundle.json file.")
+	f.StringVarP(&opts.Tag, "tag", "t", "",
```
Comment (vdice) [inline cmd/porter/bundle.go]: Shall we add an add'l example using this flag to the command examples text?

File: pkg/porter/archive.go (hunk 2 above)
Comment (jeremyrickard) [reply pkg/porter/archive.go]: i was going to add it back, eventually. just needs to be updated to work like we want it to work,

**Pull request merged**